src/components/mixins/ProxyData.js.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Gets the screen parent or null if don't have
+ * @returns {object|null}
+ */
+function findScreenOwner(control) {
+  let owner = control.$parent;
+  while (owner) {
+    const isScreen = owner.$options.name === "ScreenContent";
+    const nestedScreen =
+      owner.$parent && owner.$parent.$parent && owner.$parent.$parent.$parent;
+    const isNestedScreen =
+      nestedScreen &&
+      nestedScreen.$options._componentTag === "form-nested-screen";
+    if (isScreen && !isNestedScreen) {
+      return owner;
+    }
+    if (isNestedScreen) {
+      owner = nestedScreen;
+    } else {
+      owner = owner.$parent;
+    }
+  }
+  return null;
+}
+/**
+ * Wrap the data of a control using a Proxy
+ * @return {object} proxy
+ */
+function wrapScreenData(screen, customFunctions = null) {
+  const handler = {
+    get: (target, name) => {
+      if (name === "_parentScreen") {
+        return findScreenOwner(screen);
+      }
+      if (customFunctions && customFunctions[name]) {
+        return customFunctions[name];
+      }
+      if (name === "_parent") {
+        const screenOwner = findScreenOwner(screen);
+        // Get _parent for the current screen (e.g. Inside Loops, Inside Tabs?, RecordLists...?)
+        if (screenOwner) {
+          return wrapScreenData(screenOwner);
+        }
+        if (screen.vdata) {
+          return screen.vdata._parent;
+        }
+        return undefined;
+      }
+      // Check if vdata exists
+      if (screen.vdata !== undefined && screen.vdata !== null) {
+        return screen.vdata[name];
+      }
+      return undefined;
+    },
+    has(target, name) {
+      // customFunctions is used by RichText controls
+      // to add custom Mustache functions
+      if (screen.customFunctions && screen.customFunctions[name]) {
+        return true;
+      }
+      if (name === "_parent") {
+        return true;
+      }
+      // Check if vdata exists
+      if (screen.vdata !== undefined && screen.vdata !== null) {
+        return screen.vdata[name] !== undefined;
+      }
+      return false;
+    }
+  };
+  return new Proxy({}, handler);
```

Comment:
Just a warning here. Proxies are not being tracked by Vue reactivity list. Meaning, if something changes inside the Proxy, Vue will not track it. You have to be explicit.
Vue only tracks objects, arrays, and other stuff. If you want to track it, you have to use those or use `Vue.set` and such

Reply:
Yes that is expected (by now) since this makeProxyData() is used only internally by SelectList to get the ValidationData at:

- Get the options values
- Pmql mustache expression

@todo is to reactor this logic to access the Screen Data (validationData)

![image](https://user-images.githubusercontent.com/8028650/200431333-28db7a2f-430b-44e6-9d36-f65bd05a761a.png)

![image](https://user-images.githubusercontent.com/8028650/200431395-4194eb94-4a78-4a83-860d-239f5509a32e.png)